include/depthai/pipeline/datatype/PointCloudData.hpp

```diff
@@ -0,0 +1,205 @@
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+
+#include "depthai-shared/common/Point3f.hpp"
+#include "depthai-shared/datatype/RawPointCloudData.hpp"
+#include "depthai/build/config.hpp"
+#include "depthai/pipeline/datatype/Buffer.hpp"
+
+// optional
+#ifdef DEPTHAI_HAVE_PCL_SUPPORT
+    #include <pcl/point_types.h>
+    #include <pcl/visualization/cloud_viewer.h>
+#endif
+
+namespace dai {
+
+/**
+ * PointCloudData message. Carries ROI (region of interest) and threshold for depth calculation
+ */
+class PointCloudData : public Buffer {
+    std::shared_ptr<RawBuffer> serialize() const override;
+    RawPointCloudData& pcl;
+    std::vector<Point3f> points;
+
+   public:
+    using Buffer::getSequenceNum;
+    using Buffer::getTimestamp;
+    using Buffer::getTimestampDevice;
+
+    /**
+     * Construct PointCloudData message.
+     */
+    PointCloudData();
+    explicit PointCloudData(std::shared_ptr<RawPointCloudData> ptr);
+    virtual ~PointCloudData() = default;
+
+    std::vector<Point3f>& getPoints();
+
+    /**
+     * Retrieves instance number
+     */
+    unsigned int getInstanceNum() const;
+
+    /**
+     * Retrieves image width in pixels
+     */
+    unsigned int getWidth() const;
+
+    /**
+     * Retrieves image height in pixels
+     */
+    unsigned int getHeight() const;
+
+    /**
+     * Retrieves minimal x coordinate in milimeters
+     */
+    float getMinX() const;
+
+    /**
+     * Retrieves minimal y coordinate in milimeters
```

Comment:
mi`ll`imeter

Also, technically in depth units, which is configurable parameter in stereo node. 

Smth like:

```suggestion
     * Retrieves minimal y coordinate in depth units (millimeter by default)
```

same for all the other